frontend/src/components/InlineMessage/InlineMessage.tsx

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from 'react'
+import ReactMarkdown from 'react-markdown'
+import { Components } from 'react-markdown/src/ast-to-react'
+import { Box, Flex, Icon, useMultiStyleConfig } from '@chakra-ui/react'
+
+import { BxsErrorCircle, BxsInfoCircle } from '~/assets/icons'
+import { InlineMessageVariant } from '~/theme/components/InlineMessage'
+
+import Link from '../Link'
+
+export type InlineMessageProps = {
+  variant: InlineMessageVariant
+  children: string
+}
+
+export const InlineMessage = ({
+  variant = 'info',
+  children,
+}: any): JSX.Element => {
+  const styles = useMultiStyleConfig('InlineMessage', { variant })
+
+  const mdComponents: Components = useMemo(
+    () => ({
+      a: (props) => {
+        return <Link {...props} />
+      },
+    }),
+    [],
+  )
+
+  return (
+    <Box __css={styles.messagebox}>
+      <Flex sx={styles.item}>
+        <Icon
+          as={variant !== 'error' ? BxsInfoCircle : BxsErrorCircle}
+          __css={styles.icon}
+        />
+        <ReactMarkdown components={mdComponents}>{children}</ReactMarkdown>
```

Comment:
I realized we are forcing people to use this component with markdown. Should we make it an option instead, so users of this component can pass in anything they want, and only use markdown parser if they select that option?

Should change it for the `Banner` component too (was tunnel visioned on the way us FormSG uses banner currently, but not for all use cases)

Should we add a new prop `useMarkdown?: boolean` and only wrap with `ReactMarkdown` if that is `true`? This way, users can do this:

```tsx
return (
  <InlineMessage>
    <Box>Put anything here, does not have to be markdown</Box>
    <Input />
  </InlineMessage>
)

// Or this
return <InlineMessage useMarkdown>Now **this** is _markdown_</InlineMessage>
```

Reply:
thanks! added `useMarkdown`, see if implementation is ok